File: Domain/Workout.cs
namespace Domain
{
    public class Workout
    {
        public Guid Id { get; set; }
        public string Title { get; set; } = "";
        public string Description { get; set; } = "";
        public string Category { get; set; } = "";
        public DateTime Date { get; set; }

        public string Difficulty { get; set; } = "";

        public bool IsLiked { get; set; } = false;

        public ICollection<WorkoutAttendee> Attendees { get; set; } = new List<WorkoutAttendee>();

    }
}

File: client-app/src/app/models/workout.ts
import { Profile } from "./profile"

export interface Workout {
	id: string
	title: string
	description: string
	category: string
	date: Date
	difficulty: string
	hostUsername: string
	isLiked: boolean
	isGoing: boolean
	isHost: boolean
	host?: Profile
	attendees: Profile[]
  }
  
export class WorkoutFormValues {
	id?: string = undefined;
	title: string = '';
	description: string = '';
	category: string = '';
	date?: Date | null = null;
	difficulty: string = '';


	constructor(workout?: WorkoutFormValues) {
		if(workout) {
			this.id = workout.id;
			this.title = workout.title;
			this.description = workout.description;
			this.category = workout.category;
			this.date = workout.date;
			this.difficulty = workout.difficulty;
		}
	}
}

export class Workout implements Workout {
	constructor(init?: WorkoutFormValues) {
		Object.assign(this, init);
	}
}